engine/src/state.rs

```diff
@@ -25,6 +27,97 @@ pub struct EngineState {
     pub upgrade_delay_blocks: u64,
 }
 
+impl EngineState {
+    #[must_use]
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let borshable: BorshableEngineState = self.into();
+        borshable.try_to_vec().unwrap()
```

Comment:
Should pass on the error instead of unwrap

Reply:
Yes, it should do that. I will update it.